build/config/src/main/resources/hibernate_noorm_style.xml

```diff
@@ -0,0 +1,380 @@
+<?xml version="1.0" encoding="utf-8"?>
+<profiles version="21">
+	<profile kind="CodeFormatterProfile" name="hibernate-noorm-style" version="21">
```

Comment:
This is a good basis for our configuration, but let's not limit ourselves to it: it's only used in OGM (discontinued) and Validator (unlikely to switch to automated formatting anytime soon).

I think we can adapt the codestyle to limit the amount of changes it will lead to, i.e. to match the effective codestyle in Hibernate Search.

For example, I'd rather keep this:

```
       private static final ConfigurationProperty<Boolean> SIGNING_ENABLED =
                       ConfigurationProperty.forKey( ElasticsearchAwsBackendSettings.SIGNING_ENABLED )
                                       .asBoolean()
                                       .withDefault( ElasticsearchAwsBackendSettings.Defaults.SIGNING_ENABLED )
                                       .build();
```

Rather than formatting it like this:

```
       private static final ConfigurationProperty<Boolean> SIGNING_ENABLED = ConfigurationProperty.forKey(
                       ElasticsearchAwsBackendSettings.SIGNING_ENABLED
       )
                       .asBoolean()
                       .withDefault( ElasticsearchAwsBackendSettings.Defaults.SIGNING_ENABLED )
                       .build();
```

AFAICS, most of the questionable formatting comes from line breaking rules... Could you look into making the codestyle match the current state of the code more closely? Note you might need some changes in the plugin config as well... In particular I wonder if there's an option to prevent the plugin from removing existing line breaks... there's usually one in IDEs.

Reply:
Yeah... I'll double-check the plugin as I haven't found anything that would help us with that. I was hoping that something like:
```xml
<setting id="org.eclipse.jdt.core.formatter.join_wrapped_lines" value="false"/>
```
would help us... but that didn't work out either 😢. But I'll try to find something that works 😄 as otherwise we'll disappoint Emmanuel and/or Hardy - https://github.com/hibernate/hibernate-search/blob/58c62aea92e5c511ecf131ba9a4ca8f19fe85821/integrationtest/v5migrationhelper/engine/src/test/java/org/hibernate/search/test/dsl/SpatialDSLTest.java#L24-L26

Reply:
At first sight it looks better now, but there's still some very questionable changes... I don't know how configurable this thing is, but I'll mention what I spotted just in case you can do something about it :)

From what I can see we'd get even fewer changes if:

* line breaks were preferably inserted _before_ `implements`/`extends`/`&&`/`||`/`+`/`throws`.
* line breaks were preferably inserted outside of generic arguments (e.g. in `engine/src/main/java/org/hibernate/search/engine/backend/work/execution/OperationSubmitter.java`, `OperationSubmitter#submitToQueue`)
* `extends`/`implements` clauses split over multiple lines got automatically indented (e.g. in `engine/src/main/java/org/hibernate/search/engine/backend/document/model/spi/IndexObjectField.java`).
* generics split over multiple lines got automatically indented (e.g. in `engine/src/main/java/org/hibernate/search/engine/backend/document/model/spi/IndexObjectField.java`).
* ternary operators were preferably formatted with `?` / `:` on their own line when split over multiple lines (e.g. in `BeanResolverImpl#toSources`)
* (maybe) formatting for javadoc was disabled? I saw there's an option about that. I must admit it seems to do more harm than good, e.g. in `EngineSettings#BACKENDS`...
* Maybe more, feel free to discuss oddities on Zulip.

Reply:
> line breaks were preferably inserted before implements/extends/&&/||/+/throws.
 
this seems possible

> line breaks were preferably inserted outside of generic arguments (e.g. in engine/src/main/java/org/hibernate/search/engine/backend/work/execution/OperationSubmitter.java, OperationSubmitter#submitToQueue)

with this one it seems that it's either make this work (so that generic arguments in methods are not wrapped) or wrap things at class declaration level:
```java
public final class ElasticsearchSearchIndexScopeImpl
		extends AbstractSearchIndexScope<
				ElasticsearchSearchIndexScopeImpl,
				ElasticsearchIndexModel,
				ElasticsearchSearchIndexNodeContext,
				ElasticsearchSearchIndexCompositeNodeContext>
		implements ElasticsearchSearchIndexScope<ElasticsearchSearchIndexScopeImpl>,
		ElasticsearchSearchQueryIndexScope<ElasticsearchSearchIndexScopeImpl> {
```

> extends/implements clauses split over multiple lines got automatically indented (e.g. in engine/src/main/java/org/hibernate/search/engine/backend/document/model/spi/IndexObjectField.java).

this seems possible

> generics split over multiple lines got automatically indented (e.g. in engine/src/main/java/org/hibernate/search/engine/backend/document/model/spi/IndexObjectField.java).

Indention is controlled by one property and if we make this case work all other things get indented, so it seems it's better to leave this un-indented.

> ternary operators were preferably formatted with ? / : on their own line when split over multiple lines (e.g. in BeanResolverImpl#toSources)

this seems possible

> (maybe) formatting for javadoc was disabled? I saw there's an option about that. I must admit it seems to do more harm than good, e.g. in EngineSettings#BACKENDS...

this seems possible